pkg/kubectl/resource_printer.go

```diff
@@ -220,15 +224,15 @@ func (h *HumanReadablePrinter) validatePrintHandlerFunc(printFunc reflect.Value)
 		return fmt.Errorf("invalid print handler. %#v is not a function", printFunc)
 	}
 	funcType := printFunc.Type()
-	if funcType.NumIn() != 4 || funcType.NumOut() != 1 {
+	if funcType.NumIn() != 5 || funcType.NumOut() != 1 {
 		return fmt.Errorf("invalid print handler." +
-			"Must accept 3 parameters and return 1 value.")
+			"Must accept 5 parameters and return 1 value.")
```

Comment:
I was thinking about the same for refactoring the functions and accept an interface instead. Also maybe we should do the same for all the arguments that we are passing and wrap them into a struct, then pass only the object, writer and a formatting struct with various flags. How does it sound?